### PR TITLE
[Story of Layout] Fix race conditions

### PR DESCRIPTION
**Pull request by albertusangga**
## Changelist
- Fix race conditions when fetching the node while Flutter still does layouting
- Fix cross axis dimension calculation due to wrong refactor

### CODE REVIEW

**Comment by albertusangga** [comment]
cc @DaveShuckerow

File: packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart [hunk 1]
```diff
@@ -864,6 +888,35 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
...
   }
+
+  bool _dirty = false;
+
+  @override
+  void onFlutterFrame() {
+    if (!mounted) return;
+    if (_dirty) {
+      rateLimiter.scheduleRequest();
+      _dirty = false;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart]: move the marking as dirty as discussed in person

File: packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart [hunk 2]
```diff
@@ -864,6 +888,35 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
...
+    }
+  }
+
+  // TODO(albertusangga): investigate why onForceRefresh is not getting called
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart]: nit: end comment with a period.

File: packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart [hunk 3]
```diff
@@ -864,6 +888,35 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
...
+  }
+
+  /// Currently this is not working so we should listen to controller selection event instead.
+  @override
+  Future<void> onInspectorSelectionChanged() {
+    return null;
+  }
+
+  /// Register callback to be executed once Flutter frame is ready.
+  void marksDirty() {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart]: rename to markAsDirty

File: packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart [hunk 4]
```diff
@@ -162,7 +164,8 @@ Widget _visualizeWidthAndHeightWithConstraints({
           ),
           if (properties is FlexLayoutProperties && properties.overflowHeight)
             TextSpan(
-              text: '\nchildren takes: ${sum(properties.childrenHeights)}',
+              text:
+                  '\nchildren takes: ${toStringAsFixed(sum(properties.childrenHeights))}',
```
Comment (DaveShuckerow) [inline packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart]: Nit: 'children take', not 'children takes'.

Here and below.

**Pull request merged**